### PR TITLE
feat: replace OpenCensus telemetry with OpenTelemetry

### PR DESCRIPTION
**Pull request by Piotr-MPG**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind fix

/kind feat
> /kind build
> /kind chore
> /kind ci
> /kind docs
> /kind style
> /kind refactor
> /kind perf
> /kind test

**What this PR does / Why we need it**:
It extends the tracing experimental feature and upgrades it to OpenTelemetry (OpenCensus and OpenTracing have merged to form OpenTelemetry).

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: internal/pkg/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		httpExporter, err := otlptracehttp.New(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		options = append(options, sdktrace.WithBatcher(httpExporter))
+	}
+
+	tp := sdktrace.NewTracerProvider(options...)
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
```
Comment (hongalex) [inline internal/pkg/tracing/tracing.go:55]: I don't think `propagation.Baggage{}` is currently being used to propagate attributes across spans. Could you explain if this is necessary?

File: internal/app/server/server.go [hunk 2]
```diff
@@ -59,6 +62,26 @@ func Run(ctx context.Context, network string, cfg *config.ServiceConfig) error {
 		}),
 	}
 
+	var tracer *trace.TracerProvider
+	if cfg.EnableTrace {
+		log.Printf("Enabling CloudTrace exporter with sample rate: %f\n", cfg.ServerConfig.TraceSampleRate)
+
+		grpcOptions = append(grpcOptions,
```
Comment (hongalex) [inline internal/app/server/server.go:69]: Should we be checking for the grpc collector flag before appending these options?

File: internal/pkg/tracing/tracing.go (hunk 1 above)
Comment (Piotr-MPG) [reply internal/pkg/tracing/tracing.go]: I put it in initially thinking it'd be needed but you're right it isn't actually necessary. Removing.

File: internal/app/server/server.go (hunk 2 above)
Comment (Piotr-MPG) [reply internal/app/server/server.go:69]: Those are not for the collector but for the server, we want them as long as tracing is enabled. Although if both enableGRPCCollector and enableHTTPCollector were disabled tracing would do anything useful - which is a shortcoming but I hesitated to remove pre-existing EnableTrace. Could use stdout print as a fallback or showing a warning if tracing was enabled without collectors?

File: internal/pkg/metadb/metadb.go [hunk 3]
```diff
@@ -482,6 +523,9 @@ func (m *MetaDB) chunkObjectsSizeSum(ctx context.Context, tx *ds.Transaction, bl
 //   - NotFound: the specified record or the blobref was not found
 //   - Internal: BlobRef status transition error
 func (m *MetaDB) PromoteBlobRefToCurrent(ctx context.Context, blob *blobref.BlobRef) (*record.Record, *blobref.BlobRef, error) {
+	_, span := otel.Tracer(tracing.ServiceName).Start(ctx, "MetaDB.GetRePromoteBlobRefToCurrentcord")
```
[the remark this answers is not in the document]
Comment (Piotr-MPG) [reply internal/pkg/metadb/metadb.go]: Fixed

File: internal/pkg/config/model.go [hunk 4]
```diff
@@ -45,8 +45,10 @@ const (
 	// The following enables OpenCensus Tracing via Cloud Trace
 	// for the Datastore client library.
 	// It is EXPERIMENTAL and subject to change or removal without notice.
-	EnableTrace     = "enable_trace"
-	TraceSampleRate = "trace_sample_rate"
+	EnableTrace              = "enable_trace"
+	TraceSampleRate          = "trace_sample_rate"
+	TraceEnableGrpcCollector = "trace_enable_grpc_collector"
```
[the remark this answers is not in the document]
Comment (Piotr-MPG) [reply internal/pkg/config/model.go]: Done

File: internal/pkg/blob/gcs.go [hunk 5]
```diff
@@ -52,46 +56,68 @@ func NewBlobGCP(ctx context.Context, bucketURL string) (*BlobGCP, error) {
...
 func (b *BlobGCP) Delete(ctx context.Context, path string) error {
+	_, span := otel.Tracer(tracing.ServiceName).Start(ctx, "BlobGCP.Delete")
+	defer span.End()
+
 	return b.bucket.Delete(ctx, path)
 }
 
 // Close releases any resources used by the instance.
 func (b *BlobGCP) Close() error {
+
```
[the remark this answers is not in the document]
Comment (Piotr-MPG) [reply internal/pkg/blob/gcs.go]: removed

**Review by hongalex — APPROVED** [review]

**Pull request merged**